allennlp/models/bidaf.py

```diff
@@ -214,15 +214,10 @@ def forward(self,  # type: ignore
                        "span_end_logits": span_end_logits,
                        "span_end_probs": span_end_probs}
         if span_start is not None:
-            # Negative log likelihood criterion takes integer labels, not one hot.
-            if span_start.dim() == 2:
-                _, span_start = span_start.max(-1)
-            loss = nll_loss(util.masked_log_softmax(span_start_logits, passage_mask), span_start.view(-1))
-            self._span_start_accuracy(span_start_logits, span_start)
-            if span_end.dim() == 2:
-                _, span_end = span_end.max(-1)
-            loss += nll_loss(util.masked_log_softmax(span_end_logits, passage_mask), span_end.view(-1))
-            self._span_end_accuracy(span_end_logits, span_end)
+            loss = nll_loss(util.masked_log_softmax(span_start_logits, passage_mask), span_start.squeeze(-1))
```

Comment:
Seems odd to have to do a squeeze here still.  Is there a reasonable way to do the squeeze earlier in the data pipeline?